Clarifying documentation of custom authentication. (#785)
ParsePlatform/Docs: _includes/js/users.md

```javascript
const myAuthData = {

>   id: '12345678'  // required field. Used to uniquely identify the linked account.

Reviewer: ```suggestion
  id: '12345678'  // Required field. Used to uniquely identify the linked account.
```

```javascript
const myAuthData = {

>   id: xzx5tt123,  // The id key is required in the authData-object. Otherwise Parse server will throw the Error 252 'This authentication method is unsupported.'

Reviewer: ```suggestion
  id: xzx5tt123,  // The id key is required in the authData-object. Otherwise Parse Server will throw the Error 252 'This authentication method is unsupported'.
```

### Custom Authentication Module

Parse Server supports many [3rd Party Authenications]({{ site.baseUrl }}/parse-server/guide/#oauth-and-3rd-party-authentication).
It is possible to `linkWith` any 3rd Party Authentication by creating a custom authentication module. A custom authentication module normally consists of a client-side AuthProvider object and a back-end AuthAdapter. The client-side object should implement the [AuthProvider interface](https://github.com/parse-community/Parse-SDK-JS/blob/master/src/interfaces/AuthProvider.js). The backend AuthAdapter should implement the the functions `validateAuthData` and `validateAppId`, check out this [AuthAdapter example](https://github.com/parse-community/parse-server/blob/master/src/Adapters/Auth/AuthAdapter.js).

> When calling the `linkWith` function **without** an `authData` object the client side authenticate-method from the provider object will be called. In the other case the `authData` object will be sent directly to parse server for authentication using the backend module.

Reviewer: ```suggestion
When calling the `linkWith` function **without** an `authData` object the client side authenticate-method from the provider object will be called. In the other case the `authData` object will be sent directly to Parse Server for authentication using the backend module.
```
